src/moscot/backends/ott/solver.py

```diff
@@ -457,3 +500,199 @@ def _call_kwargs(cls) -> Tuple[Set[str], Set[str]]:
+                )
+                tgt_data_train = OTData(
+                    lin=target_split_data.data_train,
+                    condition=target_split_data.conditions_train if is_conditional else None,
+                )
+                train_dataset = OTDataset(
+                    src_data=src_data_train, tgt_data=tgt_data_train, seed=seed, is_aligned=is_aligned
+                )
+                train_loader = Loader(train_dataset, batch_size=batch_size, seed=seed)
+                src_data_validate = OTData(
+                    lin=source_split_data.data_valid,
+                    condition=source_split_data.conditions_valid if is_conditional else None,
+                )
+                tgt_data_validate = OTData(
+                    lin=target_split_data.data_valid,
+                    condition=target_split_data.conditions_valid if is_conditional else None,
+                )
+                validate_dataset = OTDataset(
+                    src_data=src_data_validate, tgt_data=tgt_data_validate, seed=seed, is_aligned=is_aligned
+                )
+                validate_loader = Loader(validate_dataset, batch_size=batch_size, seed=seed)
+                train_loaders.append(train_loader)
+                validate_loaders.append(validate_loader)
+        source_dim = self._neural_kwargs.get("input_dim", 0)
+        target_dim = source_dim
+        condition_dim = self._neural_kwargs.get("cond_dim", 0)
+        # TODO(ilan-gold): What are reasonable defaults here?
+        neural_vf = VelocityField(
+            output_dims=[*self._neural_kwargs.get("velocity_field_output_dims", []), target_dim],
+            condition_dims=(
+                self._neural_kwargs.get("velocity_field_condition_dims", [source_dim + condition_dim])
+                if is_conditional
+                else None
+            ),
+            hidden_dims=self._neural_kwargs.get("velocity_field_hidden_dims", [512, 512, 512]),
+            time_dims=self._neural_kwargs.get("velocity_field_time_dims", None),
+            time_encoder=self._neural_kwargs.get(
+                "velocity_field_time_encoder", functools.partial(time_encoder.cyclical_time_encoder, n_freqs=1024)
+            ),
+        )
+        seed = self._neural_kwargs.get("seed", 0)
+        rng = jax.random.PRNGKey(seed)
+        data_match_fn_kwargs = self._neural_kwargs.get(
+            "data_match_fn_kwargs", {"epsilon": 1e-1, "tau_a": 1.0, "tau_b": 1.0}
+        )
+        time_sampler = self._neural_kwargs.get("time_sampler", uniform_sampler)
+        optimizer = self._neural_kwargs.get("optimizer", optax.adam(learning_rate=1e-4))
+        self._solver = genot.GENOT(
+            vf=neural_vf,
+            flow=self._neural_kwargs.get(
+                "flow",
+                dynamics.ConstantNoiseFlow(0.1),
+            ),
+            data_match_fn=functools.partial(data_match_fn, typ="lin", **data_match_fn_kwargs),
```

Comment:
can we make `data_match_fn` of type `Optional[Callable[[Optional[ArrayLike], Optional[ArrayLike], Optional[ArrayLike], Optional[ArrayLike]], ArrayLike]]`
so the user can pass their own `data_match_fn`

Reply:
How would this work given the fact that the default should take arguments?  Allow both kwargs for the default and then a user-defined option?  Make the default ones not take any arguments?

Reply:
Let's expose `epsilon`,  `tau_a` and  `tau_b`, `scale_cost`, and `cost_fn`  in `data_match_fn` and the actual `solve` function, and then not provide any keyword arguments for it. 

And if `data_match_fn` is given, we say it ignores these parameters. What do you think?

Reply:
Sounds good!

Reply:
I think we'll need to expose them as before then, so no change.  `tau_a` could be used i.e., for unbalancedness.  So I'll just override use of the default function if provided.